### PR TITLE
perf(charts): improve performance on GET list

### PR DESCRIPTION
**Pull request by dpgaspar**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
This API endpoint was issuing a couple of extra queries for each row. One for `ab_user` resolving created by field and another for fetching the correct datasource. Since druid support outside of SQLAlchemy is deprecated, making an outer join with SqlaTable.

The idea to optimize and avoid a query per row, is when using `@property` has a column and the method itself references a column, make sure this column is declared on `list_columns` so it's prefetched or SQLAlchemy will issue the extra query 

Local times are:
Before:
(timing) ChartRestApi.get_list.time = 130ms - 180ms

After:
(timing) ChartRestApi.get_list.time = 25ms - 50ms

Druid charts get displayed normally but if any, an extra query is issued for each one:

Screenshot 2020-04-30 at 15 34 17

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [X] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@nytai

### CODE REVIEW

File: superset/models/slice.py [hunk 1]
```diff
@@ -72,6 +72,15 @@ class Slice(
...
     schema_perm = Column(String(1000))
     owners = relationship(security_manager.user_model, secondary=slice_user)
+    table = relationship(
+        "SqlaTable",
+        foreign_keys=[datasource_id],
+        primaryjoin="and_(Slice.datasource_id == SqlaTable.id, "
+        "Slice.datasource_type == 'table')",
+        remote_side="SqlaTable.id",
+        lazy="joined",
+    )
```
Comment (dpgaspar) [inline superset/models/slice.py:82]: Setting this relation will avoid making one extra query per row, but will not support showing the datasource for deprecated druid source (yet will issue an outer join)

Comment (dpgaspar) [reply superset/models/slice.py:82]: Updated the PR description with an example

**Comment by willbarrett** [comment]
I think Airbnb might have opinions about this. @john-bodley would this mess with your users' workflow?

**Comment by john-bodley** [comment]
@dpgaspar regarding the comment, 

> Since druid support outside of SQLAlchemy is deprecated

this isn't quite true, i.e., although we encourage environments to use Druid SQL (rather than Druid NoSQL) we still currently support Druid NoSQL and thus I'm not certain whether we should merge this PR.

I sense this PR shows the potential performance wins of actually fully deprecating the Druid NoSQL connector, i.e., there are numerous other places in the code base were the logic is complex and/or requires additional joins because there doesn't exist a foreign key between the `slices` and `tables` (or `datasources` which will be deprecated) tables.

**Comment by dpgaspar** [comment]
@john-bodley had the wrong impression regarding Druid NoSQL. 

Yet, adapted this PR so that we can have the best of both worlds. We still have the performance boost for fetching charts outside of Druid NoSQL and an extra query is issued for each Druid NoSQL existent on the query page (like before).

@willbarrett 
No user impact, just user satisfaction :)

**Review by john-bodley — APPROVED** [review]
LGTM.

**Review by willbarrett — APPROVED** [review]

**Pull request merged**